Add root node to display as parent to all assemblies

CharliePoole opened the pull request:
Fixes #73

immeraufdemhund commented on src/TestCentric/testcentric.gui/Controls/TestSuiteTreeView.cs:
```diff
@@ -599,8 +599,9 @@ public void Load(TestNode topLevelNode)
                 {
-                    foreach (var child in topLevelNode.Children)
-                        AddTreeNodes(Nodes, child, false);
+                    //foreach (var child in topLevelNode.Children)
```
remove commented out code

immeraufdemhund commented on src/TestCentric/testcentric.gui/TestCentricPresenter.cs:
```diff
@@ -391,8 +391,7 @@ public void RunFailedTests()
         {
-            EnableRunCommand(false);
-            Model.RunTests(test);
+            RunTests(new TestNode[] { test });
```
nice :) I like seeing simplified code

immeraufdemhund commented on src/TestModel/model/TestNode.cs:
```diff
@@ -54,13 +54,40 @@ public class TestNode : ITestItem
+                    break;
+            }
+
+            //IsSuite = Xml.Name == "test-suite" || Xml.Name == "test-run";
```
remove commented out code

immeraufdemhund commented on src/TestModel/model/TestNode.cs:
```diff
@@ -54,13 +54,40 @@ public class TestNode : ITestItem
-            TestCount = IsSuite ? GetAttribute("testcasecount", 0) : 1;
-            RunState = GetRunState();
+
+            switch (Xml.Name)
```
took me a second to realize this is much easier to read. Nice

immeraufdemhund replied:
i meant the entire switch. made it very clera what was to be shown.

CharliePoole commented:
I'm also going to fix the item I noted as `HACK` before merge.

CharliePoole commented on src/TestCentric/testcentric.gui/Controls/TestSuiteTreeView.cs:
```diff
@@ -1261,6 +1262,10 @@ public class TestNodeCollection : List<TestNode>, ITestItem
+            // HACK
+            if (Count == 1 && this[0].FullName == "TestRun")
+                return TestFilter.Empty;
+
```
I really don't like this and I just noticed what may be the root cause of the problem. The TestNodeCollection class is defined in the view, kused by the presenter and really belongs in the model!

CharliePoole commented:
I removed the comments you noted and also restructured. TestNodeCollection is only used in the model, so I moved it there. I took out the HACK and made the test node itself responsible (if it's a test-run node) to return an empty filter when asked for one.

I also fixed up some incorrect comments and removed code for the somewhat related TestSelection class from the solution, but kept in in the repo. Unused code that called it is compiled out using the NEW_GUI define. It would be cleaner to just delete the code, but I want to figure out how to merge the models for the old and new guis before I do that, so NEW_GUI will at least temporarily hide code in the model that is only there for the new gui.

I'd really like to be done with this and work on the new GUI!!!

immeraufdemhund commented:
sounds nice... I don't see that it got pushed up yet though.

CharliePoole commented:
Odd. I'll check.
